pallets/parachain-system/src/validate_block/implementation.rs

```diff
@@ -118,6 +118,11 @@ where
 	let inherent_data = block
 		.extrinsics()
 		.iter()
+		// Inherents are at the front of the block and are unsigned.
+		//
+		// If `is_signed` is returning `None`, we keep it safe and assume that it is "signed".
+		// We are searching for unsigned transactions anyway.
+		.take_while(|e| !e.is_signed().unwrap_or(true))
```

Comment:
@bkchr just to clarify if I get the comments correctly. 

This `take_while(..)` will include the extrinsic if `is_signed()` returns `None`, while the docs say to keep it safe and assumes the extrinsic is signed, which I understand as rejecting the extrinsic at that point instead of including it. Hence, I am confused.

Reply:
There is a `!` at the front. So, we only `take` extrinsics until we find the first signed one.

Reply:
Thanks, missed that!